console/src/main/java/org/matrix/console/Matrix.java

```diff
@@ -90,22 +91,22 @@ public synchronized MXSession getDefaultSession() {
             return sessions.get(0);
         }
 
-        ArrayList<Credentials> credsList = mLoginStorage.getCredentialsList();
+        ArrayList<HomeserverConnectionConfig> hsConfigList = mLoginStorage.getCredentialsList();
 
         // any account ?
-        if ((credsList == null) || (credsList.size() == 0)) {
+        if ((hsConfigList == null) || (hsConfigList.size() == 0)) {
             return null;
         }
 
         ArrayList<String> matrixIds = new ArrayList<String>();
         sessions = new ArrayList<MXSession>();
 
-        for(Credentials creds : credsList) {
+        for(HomeserverConnectionConfig config: hsConfigList) {
             // avoid duplicated accounts.
-            if (matrixIds.indexOf(creds.userId) < 0) {
-                MXSession session = createSession(creds);
+            if (config.getCredentials() != null || matrixIds.indexOf(config.getCredentials().userId) < 0) {
```

Comment:
Did you intend for this to be `||`? I suspect you meant `&&` as you wanted to null guard the next statement.